gpt_index/optimzation/optimizer.py

```diff
@@ -0,0 +1,102 @@
+"""Optimization related classes and functions."""
+import logging
+from typing import Dict, Optional
+
+from gpt_index.embeddings.base import BaseEmbedding
+from gpt_index.embeddings.openai import OpenAIEmbedding
+from gpt_index.indices.query.embedding_utils import get_top_k_embeddings
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.indices.response.builder import TextChunk
+
+
+class Optimizer:
+    """Optimization of a text chunk given the query by shortening the input text."""
+
+    def __init__(
+        self,
+        split_mode: str,
```

Comment:
should probably be an enum (also makes validation easier)

Reply:
Removed.